src/palace/manager/celery/tasks/playtime_entries.py

```diff
@@ -0,0 +1,379 @@
+            )
+        ]
+
+        for data_source_name in data_source_names:
+            reporting_name_with_no_spaces = (
+                f"{reporting_name}-{data_source_name}".replace(" ", "_")
+            )
+            file_name_prefix = f"{formatted_start_date}-{formatted_until_date}-playtime-summary-{reporting_name_with_no_spaces}-{uid}"
+            linked_file_name = f"{file_name_prefix}.{link_extension}"
+            # Write to a temporary file so we don't overflow the memory
+            with tempfile.NamedTemporaryFile(
+                "w+b",
+                prefix=f"{file_name_prefix}",
+                suffix=link_extension,
+            ) as temp:
+                # Write the data as a CSV
+                writer = csv.writer(
+                    TextIOWrapper(temp, encoding="utf-8", write_through=True)
+                )
+                _produce_report(
+                    writer,
+                    date_label=report_date_label,
+                    records=_fetch_report_records(
+                        session=session,
+                        start=start,
+                        until=until,
+                        data_source_name=data_source_name,
+                    ),
+                )
+
+                nested_folders = [
+                    data_source_name,
+                    "Usage Report",
+                    reporting_name,
+                    str(start.year),
+                ]
+                folder_results = google_drive.create_nested_folders_if_not_exist(
+                    folders=nested_folders,
+                    parent_folder_id=root_folder_id,
+                )
+                # the leaf folder is the last path segment in the result list
+                leaf_folder = folder_results[-1]
+
+                # store file
+                google_drive.create_file(
+                    file_name=linked_file_name,
+                    parent_folder_id=leaf_folder["id"],
+                    content_type="text/csv",
+                    stream=temp,  # type: ignore[arg-type]
+                )
+                task.log.info(
+                    f"Stored {'/'.join(nested_folders + [linked_file_name])} in Google Drive"
+                    f"{'' if not root_folder_id else f' under the parent folder (id={root_folder_id}'}."
+                )
+
+
+def _fetch_distinct_data_source_names_in_range(
+    session: Session, start: date, until: date
+) -> Query:  # type: ignore[type-arg]
+    return session.query(
+        select(
+            distinct(PlaytimeSummary.data_source_name),
+        )
+        .where(
+            and_(
+                PlaytimeSummary.timestamp >= start,
+                PlaytimeSummary.timestamp < until,
+            )
+        )
+        .order_by(PlaytimeSummary.data_source_name)
+        .subquery()
+    )
+
+
+def _fetch_report_records(
+    session: Session,
+    start: date,
+    until: date,
+    data_source_name: str,
+) -> Query:  # type: ignore[type-arg]
```

Comment:
Same as above.

Reply:
Can you fix this type hint so you don't have to add an ignore here?